scalr/resource_scalr_endpoint.go

```diff
@@ -77,11 +81,14 @@ func resourceScalrEndpointCreate(d *schema.ResourceData, meta interface{}) error
 	// Create a new options struct.
 	options := scalr.EndpointCreateOptions{
 		Name:        scalr.String(name),
-		SecretKey:   scalr.String(d.Get("secret_key").(string)),
 		Url:         scalr.String(d.Get("url").(string)),
 		Environment: environment,
 		Account:     account,
 	}
+	if SecretKey, ok := d.GetOk("secret_key"); ok {
```

Comment:
Needs to be secretKey. Title case only for struct fields you want to make public.

Also there is another `d.Get` for the same key after this if statement.